crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -2578,6 +2705,243 @@ impl CheatcodeTracer {
+                    data: data.into(),
+                    value: u256_to_revm_u256(*value),
+                    // As above, we set the gas price to 0.
+                    gas_price: rU256::from(0),
+                    gas_priority_fee: None,
+                    gas_limit: era_env.system_env.gas_limit as u64,
+                    ..env.tx
+                };
+
+                let handle = &mut storage.borrow_mut();
+                let era_db = &handle.storage_handle;
+                let mut db = era_db.db.lock().unwrap();
+
+                let result = db.call_with_evm(env).expect("failed running evm call");
+
+                let address = match result.result {
+                    revm::primitives::ExecutionResult::Success {
+                        output: revm::primitives::Output::Create(_, Some(address)),
+                        ..
+                    } => address,
+                    _ => panic!("failed deploying contract: {:?}", result.result),
+                };
+
+                tracing::info!("deployed contract address: {:?}", address);
+                let output = address.to_vec();
+
+                // update state in current EVM after performing any storage translations
+                if !result.state.is_empty() {
+                    db.commit(result.state);
+                }
+
+                self.farcall_handler.set_immediate_return(output);
+                return true
+            }
+
+            if !BROADCAST_IGNORED_CONTRACTS.contains(&current.code_address) {
+                tracing::info!("Executing in EVM mode {:?}", call);
+
+                let from = h160_to_address(current.msg_sender);
+                let (value, to) = match call {
+                    ParsedFarCall::SimpleCall { to, .. } => (rU256::ZERO, to),
+                    ParsedFarCall::ValueCall { recipient, value, .. } => {
+                        (u256_to_revm_u256(value), recipient)
+                    }
+                };
+                // let to = h160_to_address(to);
+                let calldata = call.calldata().to_vec();
+
+                let era_env = self.env.get().unwrap();
+                let mut env = into_revm_env(era_env);
+                env.block = fork_env.block.clone();
+                env.cfg.disable_eip3607 = true;
+                env.cfg.disable_base_fee = true;
+                env.cfg.disable_block_gas_limit = true;
+                env.tx = revm::primitives::TxEnv {
+                    caller: from,
+                    transact_to: revm::primitives::TransactTo::Call(h160_to_address(to)),
+                    data: calldata.into(),
+                    value,
+                    // As above, we set the gas price to 0.
+                    gas_price: rU256::from(0),
+                    gas_priority_fee: None,
+                    gas_limit: era_env.system_env.gas_limit as u64,
+                    ..env.tx
+                };
```

Comment:
you are doing the same just a few lines above

Reply:
Deduped